### PR TITLE
(FM-4615) LVM: create automated tests for filesystem

### PR DESCRIPTION
**Pull request by phongdly**

### CODE REVIEW

File: tests/beaker/tests/create_filesystem_non-existing-format.rb [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+MANIFEST
+
+step 'Inject "site.pp" on Master'
+site_pp = create_site_pp(master, :manifest => pp)
+inject_site_pp(master, get_site_pp_path(master), site_pp)
+
+step 'Run Puppet Agent to create logical volumes'
+confine_block(:except, :roles => %w{master dashboard database}) do
+  agents.each do |agent|
+    on(agent, puppet('agent -t --graph  --environment production'), :acceptable_exit_codes => [1,6]) do |result|
```
Comment (cowofevil) [inline tests/beaker/tests/create_filesystem_non-existing-format.rb]: Using ```--graph``` again. You really like that option!

File: tests/beaker/tests/create_filesystem_with_ensure_property_ext2.rb [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+MANIFEST
+
+step 'Inject "site.pp" on Master'
+site_pp = create_site_pp(master, :manifest => pp)
+inject_site_pp(master, get_site_pp_path(master), site_pp)
+
+step 'Run Puppet Agent to create logical volumes'
+confine_block(:except, :roles => %w{master dashboard database}) do
+  agents.each do |agent|
+    on(agent, puppet('agent -t --graph  --environment production'), :acceptable_exit_codes => [0,2]) do |result|
```
Comment (cowofevil) [inline tests/beaker/tests/create_filesystem_with_ensure_property_ext2.rb]: Using ```--graph``` again. You really like that option!

File: tests/beaker/tests/create_filesystem_with_param_fs_type_ext4.rb [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+MANIFEST
+
+step 'Inject "site.pp" on Master'
+site_pp = create_site_pp(master, :manifest => pp)
+inject_site_pp(master, get_site_pp_path(master), site_pp)
+
+step 'Run Puppet Agent to create logical volumes'
+confine_block(:except, :roles => %w{master dashboard database}) do
+  agents.each do |agent|
+    on(agent, puppet('agent -t --graph  --environment production'), :acceptable_exit_codes => [0,2]) do |result|
```
Comment (cowofevil) [inline tests/beaker/tests/create_filesystem_with_param_fs_type_ext4.rb]: Using ```--graph``` again. You really like that option!

File: tests/beaker/tests/create_filesystem_with_param_name_ext3.rb [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+MANIFEST
+
+step 'Inject "site.pp" on Master'
+site_pp = create_site_pp(master, :manifest => pp)
+inject_site_pp(master, get_site_pp_path(master), site_pp)
+
+step 'Run Puppet Agent to create logical volumes'
+confine_block(:except, :roles => %w{master dashboard database}) do
+  agents.each do |agent|
+    on(agent, puppet('agent -t --graph  --environment production'), :acceptable_exit_codes => [0,2]) do |result|
```
Comment (cowofevil) [inline tests/beaker/tests/create_filesystem_with_param_name_ext3.rb]: Using ```--graph``` again. You really like that option!

File: tests/beaker/tests/create_filesystem_with_param_options.rb [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+MANIFEST
+
+step 'Inject "site.pp" on Master'
+site_pp = create_site_pp(master, :manifest => pp)
+inject_site_pp(master, get_site_pp_path(master), site_pp)
+
+step 'Run Puppet Agent to create logical volumes'
+confine_block(:except, :roles => %w{master dashboard database}) do
+  agents.each do |agent|
+    on(agent, puppet('agent -t --graph  --environment production'), :acceptable_exit_codes => [0,2]) do |result|
```
Comment (cowofevil) [inline tests/beaker/tests/create_filesystem_with_param_options.rb]: Using ```--graph``` again. You really like that option!

File: tests/beaker/tests/create_filesystem_with_param_options.rb [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+logical_volume{'#{lv}':
+  ensure        => present,
+  volume_group  => '#{vg}',
+  size          => '20M',
+}
+->
+filesystem {'Create_filesystem':
+  name    => '/dev/#{vg}/#{lv}',
+  ensure  => present,
+  fs_type => 'ext2',
```
Comment (cowofevil) [inline tests/beaker/tests/create_filesystem_with_param_options.rb]: This should be ```ext4```.

File: tests/beaker/tests/create_filesystem_with_param_options.rb [hunk 7]
```diff
@@ -0,0 +1,59 @@
+require 'master_manipulator'
+require 'lvm_helper'
+require 'securerandom'
+
+test_name "FM-4615 - C96568 - create filesystem with parameter 'fs_type' and ext4 format"
```
Comment (cowofevil) [inline tests/beaker/tests/create_filesystem_with_param_options.rb]: You have two test cases with the same title. I think this one is supposed to be ```create filesystem with parameter 'options' and ext4 format```

File: tests/beaker/tests/create_filesystem_with_param_options.rb (hunk 6 above)
Comment (cowofevil) [inline tests/beaker/tests/create_filesystem_with_param_options.rb]: This should be ```ext4```.

File: tests/beaker/tests/create_filesystem_non-existing-format.rb (hunk 1 above)
Comment (phongdly) [inline tests/beaker/tests/create_filesystem_non-existing-format.rb]: haha... it's copy and paste typo. will fix it in the next push

File: tests/beaker/tests/create_filesystem_with_ensure_property_ext2.rb (hunk 2 above)
Comment (phongdly) [inline tests/beaker/tests/create_filesystem_with_ensure_property_ext2.rb]: haha... it's copy and paste typo. will fix it in the next push

File: tests/beaker/tests/create_filesystem_with_param_fs_type_ext4.rb (hunk 3 above)
Comment (phongdly) [inline tests/beaker/tests/create_filesystem_with_param_fs_type_ext4.rb]: haha... it's copy and paste typo. will fix it in the next push

File: tests/beaker/tests/create_filesystem_with_param_name_ext3.rb (hunk 4 above)
Comment (phongdly) [inline tests/beaker/tests/create_filesystem_with_param_name_ext3.rb]: haha... it's copy and paste typo. will fix it in the next push

File: tests/beaker/tests/create_filesystem_with_param_options.rb (hunk 5 above)
Comment (phongdly) [inline tests/beaker/tests/create_filesystem_with_param_options.rb]: haha... it's copy and paste typo. will fix it in the next push

**Pull request merged**